src/vector/geometry.rs

```diff
@@ -416,6 +454,61 @@ impl Geometry {
     pub fn to_geo(&self) -> Result<geo_types::Geometry<f64>> {
         self.try_into()
     }
+
+    /// Attempts to make an invalid geometry valid without losing vertices.
+    ///
+    /// Already-valid geometries are cloned without further intervention.
+    ///
+    /// Extended options are available via [`CslStringList`] if GDAL is built with GEOS >= 3.8.
+    /// They are defined as follows:
+    ///
+    /// * `METHOD=LINEWORK`: Combines all rings into a set of node-ed lines and then extracts
+    ///    valid polygons from that "linework".
+    /// * `METHOD=STRUCTURE`: First makes all rings valid, then merges shells and subtracts holes
+    ///    from shells to generate valid result. Assumes holes and shells are correctly categorized.
+    /// * `KEEP_COLLAPSED=YES/NO`. Only for `METHOD=STRUCTURE`.
+    ///   - `NO` (default):  Collapses are converted to empty geometries
+    ///   - `YES`: collapses are converted to a valid geometry of lower dimension
+    ///
+    /// When GEOS < 3.8, this method will return `Ok(self.clone())` if it is valid, or `Err` if not.
+    ///
+    /// Refer: [OGR_G_MakeValidEx](https://gdal.org/api/vector_c_api.html#_CPPv417OGR_G_MakeValidEx12OGRGeometryH12CSLConstList)
+    ///
+    /// # Example
+    /// ```rust, no_run
+    /// use gdal::vector::Geometry;
+    /// # fn main() -> gdal::errors::Result<()> {
+    /// let src = Geometry::from_wkt("POLYGON ((0 0,10 10,0 10,10 0,0 0))")?;
```

Comment:
(nit) these are probably nicer to read as below, but you don't need to change them:

```suggestion
    /// let src = Geometry::from_wkt("POLYGON ((0 0, 10 10, 0 10, 10 0, 0 0))")?;
```